src/stratigraphy/sidebar/a_above_b_sidebar.py

```diff
@@ -64,32 +65,29 @@ def depth_intervals(self) -> list[AAboveBInterval]:
         # (and includes additional lines below the actual material descriptions).
         return depth_intervals
 
-    def significant_arithmetic_progression(self) -> bool:
-        # to allow for OCR errors or gaps in the progression, we only require a segment of length 6 that is an
-        # arithmetic progression
-        segment_length = 6
-        if len(self.entries) < segment_length:
-            return self.is_arithmetic_progression()
-        else:
-            for i in range(len(self.entries) - segment_length + 1):
-                if AAboveBSidebar(self.entries[i : i + segment_length]).is_arithmetic_progression():
-                    return True
+    def close_to_arithmetic_progression(self) -> bool:
+        """Check if the depth values of the entries of this sidebar are very close to an arithmetic progressing."""
+        if len(self.entries) < 2:
             return False
 
-    def is_arithmetic_progression(self) -> bool:
-        if len(self.entries) <= 2:
-            return True
-
-        progression = np.array(range(len(self.entries)))
-        entries = np.array([entry.value for entry in self.entries])
-
-        # Avoid warnings in the np.corrcoef call, as the correlation coef is undefined if the standard deviation is 0.
-        if np.std(entries) == 0:
+        values = [entry.value for entry in self.entries]
+
+        first = values[0]
+        last = values[-1]
+        if last > first:
```

Comment:
```suggestion
        differences = [values[i + 1] - values[i] for i in range(len(values) - 1)]
        step = round(statistics.median(differences), 2)
        if step <= 0:
            return False

        arithmethic_progression = {
            # ensure we have nicely rounded numbers, without inaccuracies from floating point arithmetic
            round(value * step, 2)
            for value in range(int(first / step), int(last / step) + 1)
        }
```
improves F1 of Geoquat to 0.59796 and Zurich to 0.8911

Reply:
Geoquat indeed improved significantly. This is due to the fact that we now also consider arithmetic progressions with a step that is not a power of 10 (0.01, 0.1, 1, 10, etc.). Indeed, there are at least 4 profiles in the Geoquat validation dataset with fixed depth values at an interval of 1.5m: A1287.pdf, A1289.pdf, A1290.pdf and B627.pdf.

The Zurich F1 score was already at 89.11%, so there is no change there.

Still, the improvement for Geoquat is quite significant, thanks a lot for spotting this! I have included your proposed change in the PR now.